docs/DeveloperGuide.adoc

```diff
@@ -239,7 +239,312 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+After retrieving it, it will call the `and()` method with the User Predicate, creating a new System Predicate which will then be pushed into the top
+of the stack.
+
+image::executeNewSearchNonEmptyStack.png[width="800"]
+'''
+`revertLastSearch()` +
+
+This method will pop the System Predicate at the top of the stack. +
+In the event that the stack is already empty, this method will throw `EmptyStackException`.
+
+image::undoSearchHistoryStack.png[width="800"]
+'''
+`clearSearchHistory()` +
+
+This method will simply empty the stack.
+
+image::clearSearchHistoryStack.png[width="800"]
+
+'''
+The following sequence diagrams shows you how `find` and `undosearch` commands utilize `SearchHistoryManager` to perform Search Pruning.
+
+ - `find` command +
+
+image::SearchPruningSequenceDiagram.png[width="800"]
+
+ - `undosearch` command +
+
+image::undoSearchSequenceDiagram.png[width="800"]
+
+
+==== Design Considerations
+
+Aspect: What data is stored in search history stack +
+
+* **Alternative 1(current choice):**  Save a Stack of Predicates
+** Pros:
+    . Does not need to store the data in search history explicitly which saves memory +
+    . Any form of Search Pruning done with Predicates can reuse `SearchHistoryManager` class +
+
+** Cons:
+. Need to understand how `Predicate` works before utilizing this Class.
+. `Predicate` objects by itself does not perform the Search Pruning. In NUSSU Connect, we have to pass the `Predicate`
+   into a `FilteredList<T>` object to do the Search Pruning.
+
+* ** Alternative 2: ** Save a Stack of Lists containing Person objects in search history
+** Pros:
+    . It is easy to understand that we are filtering according to Person objects from `SearchHistoryManager` class
+
+** Cons:
+    . More memory is required as Person objects has to be duplicated multiple times into a new List.
+    . Class is inextensible to store other objects other than those from a Person classes.
+
+
+=== Add Skill Level Command
+
+==== Current Implementation
+
+The add skill mechanism builds on the `addressBookParser`. This as well
+as it's subclass `addSkillCommandParser` ensures that the correct number of arguments
+is given to the command.
+
+The following shows how the application handles the request to change a skill in one particular scenario.
+
+Step 1. The user launches the application. The application boots up and lists all members.
+
+Step 2. The user locates the person he wants to add on at Index 4. They execute the `asl 4 s/Photography l/60` command.
+
+Step 3.'LogicManager' calls the 'parseCommand' in the 'AddressBookParser', which calls `AddSkillCommandParser` to
+parse it.
+
+Step 4. After parsing, the command is sent to the `Model` which alters the `Person` object by modifying their
+`Skill` and `SkillLevel` properties.
+
+Step 5. The result is encapsulated as a `CommandResult` object which is passed back to the `UI`.
+
+
+Before executing the command:
+
+image::aslbefore.png[width="800"]
```

Comment:
Again the before and after snapshots of the UI are not mandatory here. Perhaps you could include them in the **User guide**. You may try to draw a UML diagram to describe your design. Note that each team member should provide at least 1 feature accompanied by 1 UML diagram.